app/Http/Controllers/ImageController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Resources\CategoryResource;
+use App\Models\Category;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+use App\Http\Requests\CategoryRequest;
+
+class ImageController extends Controller
+{
+    public function uploadIcon(CategoryRequest $request)
+    {
+        $categoryId = $request->input('category_id');
+        $category = Category::find($categoryId);
+
+        if (!$category) {
+            Log::error('Category not found');
+            return response()->json(['error' => 'Category not found'], 404);
```

Comment:
@Geri-PG You can use route model binding to handle 404 error in the background instead of manually handling it (in the function add param Category $category). If the file is uploaded, update icon field for the category record